docs/MessageTokenizer.md

```diff
@@ -0,0 +1,260 @@
+>
+> Throws: IllegalStateException
+
+```java
+  if(!this.hasNextLine()) {
+      throw new IllegalStateException("No more lines found!");
+    }
+```
+
+Otherwise, grabs the next line in the sequence.
+
+### hasNextPattern(Pattern)
+
+> Return Type: boolean
+> 
+> Throws: Nothing
+
+`true` if `hasNext()` is true **and** the given pattern has a match in the remaining content.
+
+### nextPattern(Pattern)
+
+> Return Type: MessageTokenizer.Token
+>
+> Throws: IllegalStateException
+
+```java
+    if(!this.hasNextRegex(pattern)) {
+      throw new IllegalStateException("No more occurrences found!");
+    }
+```
+```java
+      if(!matcher.find()) {
+        throw new IllegalStateException("Couldn't find any matches!");
+      }
+```
+ 
+ Otherwise, returns the regex in the content and steps to the end of it.
+
+### hasNextInvite()
+
+> Return Type: boolean
+>
+> Throws: Nothing
+
+`true` if there's an invite in the message, otherwise `false`.
+
+### nextInvite()
+
+> Return Type: MessageTokenizer.InviteToken
+> 
+> Throws: IllegalStateException
+
+```java
+    if(!this.hasNextInvite()) {
+      throw new IllegalStateException("No more invites found!");
+    }
+```
+
+```java
+      if(!matcher.find()) {
+        throw new IllegalStateException("Couldn't find any matches!");
+      }
+```
+
+Otherwise, returns the next Invite found (Example: `discord.gg/NxGAeCY`) and steps to the end of it.
+
+### hasNextMention()
+
+> Return Type: boolean
+>
+> Throws: Nothing
+
+`true` if there is another mention (channel, user, role, etc..) in the message, otherwise `false`
+
+### nextMention()
+
+> Return Type: MessageTokenizer.MentionToken
+>
+> Throws: IllegalStateException
+
+```java
+    if(!this.hasNextMention()) {
+      throw new IllegalStateException("No more mentions found!");
+    }
+```
+
+```java
+      } else {
+        throw new IllegalStateException("Couldn't find a mention even though it was found!");
+      }
+```
+
+Otherwise, returns a `MessageTokenizer.RoleMentionToken`, `MessageTokenizer.UserMentionToken`, or a `MessageTokenizer.ChannelMentionToken`.
+
+### hasNextEmoji()
+
+> Return Type: boolean
+>
+> Throws: Nothing
+
+`true` if there is an emoji in the remaining content, otherwise `false`
+
+### nextEmoji()
+
+> Return Type: MessageTokenizer.CustomEmojiToken
+>
+> Throws: IllegalStateException
+
+```java
+    if(!this.hasNextEmoji()) {
+      throw new IllegalStateException("No more custom server emojis found!");
+    } 
+```
+
+Otherwise, returns the emoji in the sequence.
+
+
+### hasNextUnicodeEmoji(Emoji)
+
+> Return Type: boolean
+>
+> Throws: Nothing
+
+`true` if the unicode exists in the remaining content, otherwise `false`
+
+### nextUnicodeEmoji(Emoji)
+
+> Return Type: MessageTokenizer.UnicodeEmojiToken
+>
+> Throws: Nothing
+
+Returns the next specified unicode emoji in the sequence.
+
+### getContent()
+
+> Return Type: String
+>
+> Throws: Nothing
+
+Returns the content of the `MessageTokenizer`.
```

Comment:
Does this return the content from the beginning or from where it currently is at?